prov/sm2/src/sm2.h

```diff
@@ -119,7 +120,7 @@ struct sm2_rx_entry {
 };
 
 struct sm2_tx_entry {
-	struct sm2_cmd cmd;
+	struct sm2_free_queue_entry cmd;
```

Comment:
deleted sm2_tx_entry b/c it isn't needed.

Each endpoint has a free queue (your naming and Nemesis's naming align here), and that queue is initialized with a finite number of free queue entries.  The free queue entry holds the sm2 assigned header and the user's message in SharedMemory, and can be placed on a peers recv queue.

Command (cmd) doesn't add any value b/c we are not commanding anything.  sm2_cmd is nice because it has 6 letters instead of 20.  I would happily change `sm2_free_queue_entry` to `sm2_fqe` for conciseness.  I think the word we are really looking for is a wrapper around a header + message.  In this sense, I think fqe does a better job than command, b/c it is clear that the free queue entry lives on the free queue.

Also, in SM2, the sender owns the FQ, FQE whereas in SHM, the receiver owns the CMD + CMD QUEUE.

Reply:
switching to `sm2_fqe`